File: src/main/kotlin/FindLauncher.java
import org.kohsuke.args4j.Argument;
import org.kohsuke.args4j.CmdLineException;
import org.kohsuke.args4j.CmdLineParser;
import org.kohsuke.args4j.Option;

public class FindLauncher {

    @Option(name = "-d", usage = "Directory")
    private String directory;
    private Boolean isD;

    @Option(name = "-r", usage = "Find in subdirectories")
    private Boolean subDirectories;

    @Argument(required = true, metaVar = "InputName", usage = "File to be found")
    private String InputFileName;

    public static void main(String[] args) {
        new FindLauncher().launch(args);
    }
    private void launch(String[] args) {
        CmdLineParser parser = new CmdLineParser(this);

        try {
            parser.parseArgument(args);
        } catch (CmdLineException e) {
            System.err.println(e.getMessage());
//            System.err.println(e.getMessage());
            parser.printUsage(System.err);
            throw new IllegalArgumentException("");
        }
//        FindK find = new FindK(InputFileName, directory);
        FindKt.start(InputFileName, directory, subDirectories, isD);
    }
}

File: src/main/kotlin/FindLauncher.kt
import org.kohsuke.args4j.Argument
import org.kohsuke.args4j.CmdLineException
import org.kohsuke.args4j.CmdLineParser
import org.kohsuke.args4j.Option
import java.io.File

class FindLauncher {
    @Option(name = "-d", usage = "Directory")
    private var directory = File("").absolutePath

    @Option(name = "-r", usage = "Find in subDirectories")
    private var subDirectories = false

    @Argument(required = true, metaVar = "InputName", usage = "File to be found")
    private var inputFileName: String = ""


    private fun launch(args: Array<String>) {
        val parser = CmdLineParser(this)
        try {
            parser.parseArgument(*args)
        } catch (e: CmdLineException) {
            System.err.println(e.message)
            System.err.println("find [-r] [-d directory] filename.txt")
            parser.printUsage(System.err)
            throw IllegalArgumentException("")
        }
        val find = start(subDirectories, directory, inputFileName)
        if (find.isEmpty()) println("File not found")
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            FindLauncher().launch(args)
        }
    }
}